src/shogun/base/SGObject.h

```diff
@@ -1286,6 +1283,22 @@ class CSGObject
 		int64_t m_next_subscription_index;
 	};
 
+template <class T>
+T* make_clone(T* orig, ParameterProperties pp = ParameterProperties::ALL)
+{
+	REQUIRE(orig, "No object provided.\n");
+	auto clone = orig->clone(pp);
+	ASSERT(clone);
+	return static_cast<T*>(clone);
+}
+
+template <class T>
+const T* make_clone(const T* orig, ParameterProperties pp = ParameterProperties::ALL)
+{
+	REQUIRE(orig, "No object provided.\n");
+	return orig->clone(pp)->template as<T>();
```

Comment:
Shouldn't you also do a static_cast here?

Reply:
yes! updated.